src/com/palmergames/bukkit/towny/object/Town.java

```diff
@@ -432,8 +434,13 @@ public double getTownBlockCostN(int inputN) throws TownyException {
 		double nextprice = getTownBlockCost();
 		int i = 1;
 		double cost = nextprice;
+		boolean hasmaxprice = TownySettings.getMaxClaimPrice() != -1;
+		double maxprice = TownySettings.getMaxClaimPrice();
 		while (i < inputN){
-			nextprice = Math.round(Math.pow(TownySettings.getClaimPriceIncreaseValue() , getTownBlocks().size()+i) * TownySettings.getClaimPrice());			
+			if(hasmaxprice && nextprice > maxprice)
+				nextprice=maxprice;
+			else
+				nextprice = Math.round(Math.pow(TownySettings.getClaimPriceIncreaseValue() , getTownBlocks().size()+i) * TownySettings.getClaimPrice());
```

Comment:
This change has some logic errors regarding to the max price. Since the conditional is first thing in the loop, that means that the loop will always be checking the previous iteration's value of next price. This provides the opportunity for the previous iteration's `nextprice` to be above max price and not be adjusted. 

Also resetting the `nextprice` to `maxprice` causes some issues as well because it will fail the conditional since `nextprice` will be equal to `maxprice` and not greater than `maxprice`. That means it will take the else branch and force a recalculation that recalculation could be greater than the `maxprice`.

There are quite a few ways to go about fixing this, so it's up to you for whatever approach you want to take. Just make sure to validate that `cost` will never add a value that is greater than `maxprice` if it exists.

Reply:
I think you could:
```
nextprice = Math.round(Math.pow(TownySettings.getClaimPriceIncreaseValue() , getTownBlocks().size()+i) * TownySettings.getClaimPrice());

if(hasmaxprice)
	nextprice = Math.min(nextprice, maxprice);

cost += nextprice;
i++;
```
no?

Reply:
I made it
`nextprice = Math.round(Math.pow(TownySettings.getClaimPriceIncreaseValue() , getTownBlocks().size()+i) * TownySettings.getClaimPrice());
			
			if(hasmaxprice && nextprice > maxprice) {
				cost += maxprice * ( inputN - i);
				break;
			}
			
			cost += nextprice;
			i++;`
now so that the loop doesn´t keep running for no reason